firebase-database/src/main/java/com/google/firebase/database/connection/PersistentConnectionImpl.java

```diff
@@ -344,6 +346,21 @@ public void listen(
     doIdleCheck();
   }
 
+  @Override
+  public Task<Object> get(List<String> path, Map<String, Object> queryParams) {
+    QuerySpec query = new QuerySpec(path, queryParams);
+    TaskCompletionSource<Object> source = new TaskCompletionSource<>();
+    Task<Object> task;
+    if (connected()) {
+      task = sendGet(query, source);
+    } else {
+      source.setException(new Exception("Client offline!"));
```

Comment:
Please drop exclamation mark as this is user facing. Maybe make this "Client is offline".